lambda/initializer.js

```diff
@@ -8,24 +8,86 @@ const defaultPowerValues = process.env.defaultPowerValues.split(',');
  */
 module.exports.handler = async(event, context) => {
 
-    const {lambdaARN, num} = event;
-    const powerValues = extractPowerValues(event);
+    const {
+        lambdaARN,
+        num,
+        powerValues,
+        onlyColdStarts,
+    } = extractDataFromInput(event);
 
     validateInput(lambdaARN, num); // may throw
 
     // fetch initial $LATEST value so we can reset it later
-    const initialPower = await utils.getLambdaPower(lambdaARN);
+    const {power, envVars} = await utils.getLambdaPower(lambdaARN);
+
+    let lambdaFunctionsToSet = [];
 
     // reminder: configuration updates must run sequentially
     // (otherwise you get a ResourceConflictException)
-    for (let value of powerValues){
-        const alias = 'RAM' + value;
-        await utils.createPowerConfiguration(lambdaARN, value, alias);
+    for (let powerValue of powerValues){
+        const baseAlias = 'RAM' + powerValue;
+        if (!onlyColdStarts){
+            lambdaFunctionsToSet.push({lambdaARN: lambdaARN, powerValue: powerValue, envVars: envVars, alias: baseAlias});
+        } else {
+            for (let n of utils.range(num)){
+                let alias = utils.buildAliasString(baseAlias, onlyColdStarts, n);
+                const currentEnvVars = {
+                    ...envVars,
+                };
+                // set Env Var to a unique value to force version publishing
+                currentEnvVars.LambdaPowerTuningForceColdStart = alias;
+                // here we inject a custom env variable to force the creation of a new version
+                // even if the power is the same, which will force a cold start
+                lambdaFunctionsToSet.push({lambdaARN: lambdaARN, powerValue: powerValue, envVars: currentEnvVars, alias: alias});
+            }
+        }
     }
+    lambdaFunctionsToSet.push({lambdaARN: lambdaARN, powerValue: power, envVars: envVars});
 
-    await utils.setLambdaPower(lambdaARN, initialPower);
+    const returnObj = {
+        initConfigurations: lambdaFunctionsToSet,
+        iterator: {
+            index: 0,
+            count: lambdaFunctionsToSet.length,
+            continue: true,
+        },
+        powerValues: powerValues
+    }
+    return returnObj;
+};
 
-    return powerValues;
+module.exports.versionPublisher = async(event, context) => {
```

Comment:
ideally this would be in a new file, to keep the one-function-per-file convention :)

Reply:
I will fix this